bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -779,6 +810,20 @@ private void internalRemove(final Object[] elements) {
 		}
 	}
 
+	/**
+	 * Returns the data of the last item on the viewer.
+	 *
+	 * @return may return null
+	 */
+	Object getLastElement() {
+		Item[] items = doGetItems();
```

Comment:
This does too much. Instead of trying to get all elements, we should directly check for the last one